docs/users/installation.md

```diff
@@ -190,6 +191,41 @@ Both the `scalafix` & the `scalafixAll` task aggregate like the `compile`
 and `test` tasks. To run Scalafix on all projects for both main and test
 sources you can execute `scalafixAll`.
 
+### Run scalafix automatically on compile
+
+If you set `scalafixOnCompile` to `true`, the rules declared in `.scalafix.conf`
+(or in the file located by `scalafixConfig`) will run automatically each time
+`compile` is invoked either explicitly or implicitly (for example when
+executing `test` or `publishLocal`). Lint errors will fail that invocation,
+while rewrites will be applied.
+
+Although this looks like an easy way to use Scalafix as a linter, use this
+feature with care as it has several shortcomings, for example:
+
+1. Some rules such as `RemoveUnused` can be counter-productive if applied too
+   often/early, as the work-in-progress code that was just added might disappear
+   after a simple `test`.
+1. If you run many semantic rules by default, the last one(s) to run might see
+   stale information and fail the invocation, which needs to be re-run manually.
+   This is [not specific to `scalafixOnCompile`](https://github.com/scalacenter/scalafix/issues/1204),
+   but the problem becomes much more visible with it.
+1. To keep the overhad minimal, `scalafixCaching` is automatically enabled when
+   `scalafixOnCompile` is, which can cause unexpected behaviors if you run into
+   false positive cache hits. `scalafixCaching` can explicitly be set to
+   `false` in that case.
+1. Non-idempotent rewrite rules might get you in an infinite loop where sources
+   never converge - not specific to `scalafixOnCompile` either, but rather
+   confusing when triggered automatically.
+1. Bugs in rule implementations can prevent you from getting a successul
+   `compile`, blocking testing or publishing for example
+
+Some of these shortcomings can be mitigated by the `scalafixOnCompileConfig`
+setting key, which allows to use a separate configuration file whenever
+Scalafix is triggered by `compile`. This way, Scalafix can execute a
+different, safer set of rules compared to the regular `scalafixConfig` that
+remains used for explicit `scalafix` & `scalafixAll` invocations.
```

Comment:
@taisukeoe I have (shamelessly) included your own concern in this list. Feedback very much welcomed!

Reply:
@bjaglin Great capture 👍  Thank you very much!

In the future, it would be nicer to describe what kind of rules are recommended for on-compile, although I haven't found the adequate words yet. 

The safest way is to recommend only diagnostic rules, while there are few rules only for diagnostics. I feel it is a bit too conservative and might miss some useful combinations.

So at this moment, I'd agree with your explanation.